src/main/kotlin/org/axonframework/intellij/ide/plugin/api/MessageHandlerType.kt

```diff
@@ -0,0 +1,72 @@
+package org.axonframework.intellij.ide.plugin.api
+
+import com.intellij.openapi.project.Project
+import com.intellij.psi.JavaPsiFacade
+import com.intellij.psi.PsiClass
+import com.intellij.psi.search.GlobalSearchScope.allScope
+import com.intellij.psi.search.searches.AnnotatedElementsSearch
+import com.intellij.psi.util.CachedValue
+import com.intellij.psi.util.CachedValueProvider
+import com.intellij.psi.util.CachedValuesManager
+import com.intellij.psi.util.PsiModificationTracker
+import java.util.concurrent.ConcurrentHashMap
+
+enum class MessageHandlerType(
+        val messageType: MessageType?,
+        val annotationName: String,
+) {
+    COMMAND(MessageType.COMMAND, "org.axonframework.commandhandling.CommandHandler"),
+    EVENT(MessageType.EVENT, "org.axonframework.eventhandling.EventHandler"),
+    SAGA(MessageType.EVENT, "org.axonframework.modelling.saga.SagaEventHandler"),
+    EVENT_SOURCING(MessageType.EVENT, "org.axonframework.eventsourcing.EventSourcingHandler"),
+    QUERY(MessageType.QUERY, "org.axonframework.queryhandling.QueryHandler"),
+    COMMAND_INTERCEPTOR(MessageType.COMMAND, "org.axonframework.modelling.command.CommandHandlerInterceptor"),
```

Comment:
We can definitely support them, but I think we should consider whether we want that, and how. Personally, I would like the information I get from the plugin to be as small as possible, but still useful. Adding an `@ExceptionHandler` reference to every command handler annotation would make it feel bloated very quickly.

In order to synchronize our thoughts a bit, I will dump here how the plugin works and why it works this way. I will also add it to documentation somewhere. Currently, the plugin works like this:

It scans for all methods with a handler annotation (including meta-annotations, 1 level down). It then resolves the payload and message type, depending on the annotation and location

The line markers use this information at handler methods to show the publishers. The publishers are found by finding constructor references of the message. Abstract classes and interfaces are supported, so if you have an interface in your event handler it will show all message constructors it could handle. 

The other way around, when a constructor invocation is found the plugin checks for message handlers with that class. If there are any, it drops a linemarker with all handlers (again, it does understand inheritance). 

At first, the plugin scanned for publishers (EventGateway, AggregateLifecycle, etc), but that came with a huge limitation; you have to construct messages and send them to the publisher directly. In the codebase at PoR we actually had a list of classes that could create events, return them as a list and the aggregate then looped over and applied them. The plugin cannot handle that if we act solely based on publishers (it would try to scan for Any, or the command interface). Also, I have found scanning for constructors to be very quick in IntelliJ, but scanning for method references of publishers to be very slow. 



ResetHandlers for example, do have a payload, but it will generally not be constructed by the developers. Only by the framework using the reset methods. 
I do see a use-case: On classes, we can show a line marker with relevant Axon stuff. For example; 
- Show command interceptors on classes with `@Aggregate`  or `@Entitity` annotation
- Show reset handlers on classes that define a reset handler

I don't know enough about deadlines to think of a way to implement them, I'll research a bit and perhaps we can talk about a nice way to support it.

Reply:
Thanks for the thorough explanation here, @Morlack. 
I agree, there's no merit in having the `@ExceptionHandler` here, simply because it would become too bloated.

> I don't know enough about deadlines to think of a way to implement them, I'll research a bit and perhaps we can talk about a nice way to support it.

Be sure to hit me up if you need a primer on this to help you further.
Or, make an issue, and I'll resolve it once this PR is in! ;-)